cmd/probe/eventhandler.go

```diff
@@ -54,7 +54,29 @@ func (pe *ProbeEvent) addDiskEvent(msg controller.EventMessage) {
 		}
 		glog.Info("processed data for ", diskDetails.ProbeIdentifiers.Uuid)
 		oldDr := pe.Controller.GetExistingResource(diskList, diskDetails.ProbeIdentifiers.Uuid)
-		pe.Controller.PushDiskResource(oldDr, diskDetails)
+		// if old DiskCR doesn't exist and parition is found, it is ignored since we don't need info
+		// of partition if disk as a whole is ignored
+		if oldDr == nil && len(diskDetails.PartitionData) != 0 {
```

Comment:
The functionality mentioned is currently done only by `path-filter`. That is, for an exclude config of `/dev/sda`, path-filter will exclude `/dev/sda`,`/dev/sda1`, `/dev/sda2` etc. But for `os-disk-filter` , it will exclude only `/dev/sda`. The functionality to exclude all matching devices  has to be implmented in the `os-disk-filter`.  Any further changes needed on the `os-filter` @shovanmaity